types/tls_altnames.pp

```diff
@@ -1,7 +1,9 @@
 # @summary a type to describe TLS alternative names in certificates
 type K8s::TLS_altnames = Array[
-  Variant[
-    Stdlib::Fqdn,
-    Stdlib::IP::Address::Nosubnet,
+  Optional[
+    Variant[
+      Stdlib::Fqdn,
+      Stdlib::IP::Address::Nosubnet,
+    ]
   ]
 ]
```

Comment:
This is not valid, you can't have undef values in the array. Perhaps you mean `Optional[Array[...` - though that would need changes in `k8s::server::tls::cert` to support `undef` instead of just a regular empty array.